scala/scala.bzl

```diff
@@ -384,3 +424,193 @@ def scala_repositories():
     url = "https://oss.sonatype.org/content/groups/public/org/scalatest/scalatest_2.11/2.2.6/scalatest_2.11-2.2.6.jar",
     sha256 = "f198967436a5e7a69cfd182902adcfbcb9f2e41b349e1a5c8881a2407f615962",
   )
+
+
+# TODO is it possible to share scala-library with the above?
+def scrooge():
+  native.maven_jar(
+    name = "libthrift",
+    artifact = "org.apache.thrift:libthrift:0.8.0"
+  )
+
+  native.maven_jar(
+    name = "scrooge_core",
+    artifact = scala_mvn_artifact("com.twitter:scrooge-core:4.6.0")
+  )
+
+  native.maven_jar(
+    name = "finagle_thrift",
+    artifact = scala_mvn_artifact("com.twitter:finagle-thrift:6.34.0")
+  )
+
+  #scrooge-generator related dependencies
+  native.maven_jar(
+    name = "scrooge_generator",
+    artifact = scala_mvn_artifact("com.twitter:scrooge-generator:4.6.0")
+  )
+  native.maven_jar(
+    name = "scopt",
+    artifact = scala_mvn_artifact("com.github.scopt:scopt:3.3.0")
+  )
+  native.maven_jar(
+    name = "twitter_util",
+    artifact = scala_mvn_artifact("com.twitter:util-core:6.33.0")
+  )
+  native.maven_jar(
+    name = "util_logging",
+    artifact = scala_mvn_artifact("com.twitter:util-logging:6.33.0")
+  )
+  native.maven_jar(
+    name = "scala_parser_combinators",
+    artifact = scala_mvn_artifact("org.scala-lang.modules:scala-parser-combinators:1.0.4")
+  )
+
+
+_thrift_filetype = FileType([".thrift"])
+_scroogelib_tar_filetype = FileType([".scroogelib.tar"])
+
+#TODO what even is the output of this? I believe it is a archive with the thrifts? seems right...
+def _scrooge_library_impl(ctx):
+  # need to deal with transitive archive files etc
+  ctx.action(
+    inputs = ctx.files.srcs,
+    outputs = [ctx.outputs.libarchive],
+    # tar -c src/thrift/com/example/thrift/AStruct.thrift > scrooge_thrifts.scroogelib.tar
+    command = "echo yeah yeah yeah", #TODO do something real :P
+  )
+  transitive_archive_files = []
+  return struct(
+    srcs=ctx.files.srcs,
+    _transitive_archive_files=transitive_archive_files,
+  )
+
+def _gen_scrooge_srcjar_impl(ctx):
+  #TODO would there be a case in which we want/need to allow more jars here?
+  transitive_cjars = [ctx.files._scrooge_core, ctx.files._libthrift, ctx.files._finagle_thrift, ctx.files._scalalib]
+  transitive_cjars = [file for files in transitive_cjars for file in files]
```

Comment:
I will try to get it to work now that I understand how it works a bit better. I was failing to get it to work but maybe it works now. If you can scan it and given pointers on how to restructure it, this is definitely an area where I "yolo'd" a bit